app/modules/enduser_setup.c

```diff
@@ -680,8 +775,8 @@ static void serve_status (struct espconn *conn)
  */
 static void enduser_setup_http_disconnect(struct espconn *espconn)
 {
-  ENDUSER_SETUP_DEBUG(lua_getstate(), "enduser_setup_http_disconnect");
-//TODO: Construct and maintain os task queue(?) to be able to issue system_os_task with espconn_disconnect.
+  ENDUSER_SETUP_DEBUG("enduser_setup_http_disconnect");
+  task_post_high(do_disconnect_tcp_handle, (task_param_t) espconn);
```

Comment:
This is not a safe approach. The `espconn *` here is frequently a temporary lwIP *internal* struct (when it is passed in to an API callback), and is not valid after the callback returns. To do the disconnects safely, you'll need to stash away the `remote_ip` and `remote_port` like what is done in the `scan_listeners`. Apologies for not noticing this earlier.